### ISSUE TITLE
composer install --no-cache flag for multiple simultaneously running installs

### ISSUE THREAD

**Issue by meatch** [composer/composer#9758]
### Overview
**Goal:** Run multiple composer install processes simultaneously
**Why:** speeding up deployment and installation multiple apps, single system.
**Problem:** When these apps share packages it looks like composer is telling one process it's cached because they are running simultaneously.

Would be great to run: `composer install --no-cache`, where composer will not use cache for that install.

### Sample Breaking
Here is an example where it appears to try and leverage cache. I also get it could be related to case sensitivity as the error implies, however, when I run each app `composer install` individually there are no issues.

```
The process "unzip -qq  '/var/www/html/web/qrez/montrose-agent-front-end/vendor/zendframework/zendframework1/b0908b51eb296acd5197b78c429c81b3' -d '/var/www/html/web/qrez/montrose-agent-front-end/vendor/composer/8cd73553'" exceeded the timeout of 300 seconds.
    The archive may contain identical file names with different capitalization (which fails on case insensitive filesystems)
    Unzip with unzip command failed, falling back to ZipArchive class
    The process "unzip -qq  '/var/www/html/web/qrez/montrose-agent-back-end/vendor/monolog/monolog/0513d609c07f47c99e9478fb677d0a5e' -d '/var/www/html/web/qrez/montrose-agent-back-end/vendor/composer/5f114443'" exceeded the timeout of 300 seconds.
    The archive may contain identical file names with different capitalization (which fails on case insensitive filesystems)
    Unzip with unzip command failed, falling back to ZipArchive class
```

### Link to Closed Issue (same request, but for a different reason)
Link to another ticket that was closed. Similar request, but was deemed was not a valid reason to change the code. Perhaps my request demonstrates a more modern use case.

https://github.com/composer/composer/issues/1896

**Comment by Seldaek** [comment]
Composer does have --no-cache since 1.9.0, is that not working for you?

Alternatively you can also set COMPOSER_CACHE_DIR env var to a unique temp dir  for every parallel process you run, so they have a one-off cache dir which could get wiped after.

However, with the latest version you should theoretically not have problems running things in parallel as we do atomic cache writes. I can't say the use case has seen a lot of usage though so it's possible things are broken.

As for the error log you posted, it seems it failed after 300sec of timeout, which may be due to overloaded disk if you run many installs in parallel? Or I am not sure what else could cause this.

**Comment by meatch** [comment]
Thanks @Seldaek, very helpful. 

> Composer does have --no-cache since 1.9.0, is that not working for you?

I did not realize that it had been added, I think I must have gotten discouraged by the last thread that was closed and verdict to not implement. I will try this out and see how it goes.

> COMPOSER_CACHE_DIR 

This also sounds like it may be a good options as well!

Thank you for the assist.

**Comment by Seldaek** [comment]
👍🏻 Ok great, closing this already but feel free to post again if needed.
